src/manager.cr

```diff
@@ -62,7 +75,13 @@ class HTTPSession
     end
 
     private def new_session_id
-      random.urlsafe_base64
+      100.times do
+        new_id = random.urlsafe_base64(session_id_length)
+        # ensure uniqueness
+        return new_id unless @storage[new_id]
```

Comment:
You may want to introduce a method to the `StorageInterface` specifically to test for the presence of a key. For a memory backed storage it makes no difference (I think), but for a database/redis/whatever retrieving the session could be expensive.

An alternative would be a method in the interface to atomically create (and ensure) a unique new key (maybe with a block parameter to create new keys?). It's not a problem right now because Crystal is single threaded, but you may have a race condition in `#set` between creating a new session ID and saving the session in it.

Reply:
I tried to move a bit into that direction with transferring ID generation to the storage backend.

This allows more specialized backends to use intrinsic ID generation mechanisms if available.

That should be plenty for now. Collissions are super rare. If they weren't, we'd have more of a attack vector than a consistency issue.

Reply:
Nice! I like the changes, but the `new_session_id` in the `Manager` feels a bit out of place. Maybe extract it to a module (which can then be included from `Storage::Memory` and other implementations)? Anyway, it's nitpicking at this point.

Reply:
Yeah, this could perhaps use some polishing. But it's just a helper method that needs to be placed somewhere.